cmake/std/atdm/ATDMDevEnvSettings.cmake

```diff
@@ -293,7 +293,16 @@ ATDM_SET_CACHE(Kokkos_ENABLE_CXX11_DISPATCH_LAMBDA ON CACHE BOOL)
 ATDM_SET_CACHE(Kokkos_ENABLE_CUDA_LAMBDA "${ATDM_USE_CUDA}" CACHE BOOL)
 ATDM_SET_CACHE(Kokkos_ENABLE_DEBUG_BOUNDS_CHECK "${Trilinos_ENABLE_DEBUG}" CACHE BOOL)
 ATDM_SET_CACHE(Kokkos_ENABLE_DEBUG "${Trilinos_ENABLE_DEBUG}" CACHE BOOL)
-ATDM_SET_CACHE(KOKKOS_ARCH "$ENV{ATDM_CONFIG_KOKKOS_ARCH}" CACHE STRING)
+
+# Update Kokkos arch, expected ATDM_CONFIG_KOKKOS_ARCH=arch1;arch2;arch3
+# set the modern Kokkos arch, e.g., KOKKOS_ARCH=arch1 => Kokkos_ARCH_arch1=ON
+set(kokkos_arch_list "$ENV{ATDM_CONFIG_KOKKOS_ARCH}")
+foreach(kokkos_arch_loop_var ${kokkos_arch_list} )
+  ATDM_SET_ENABLE(Kokkos_ARCH_${kokkos_arch_loop_var} ON)
```

Comment:
@jjellio What's the difference between ATDM_SET_ENABLE and ATDM_SET_CACHE? Will this work if you rerun cmake with a different ``$ATDM_CONFIG_KOKKOS_ARCH``?

Reply:
Don't know. If I use SET_CACHE, it doesn't print the nice little message 'setting foo-bar'. They use set_enable later.

I've had this patch for a while, It's needed on ARM since they configure your BLAS library's architecture from the compiler's architecture flags.  I saw your bug report and decided to commit it.

It'd be nice if the cmake gurus can explain the way it should be done.

Reply:
It's also not obvious if I need to create the list. I had originally tried without the list variable, but it didn't work. It's just really annoying when this doesn't work.

Reply:
> What's the difference between ATDM_SET_ENABLE and ATDM_SET_CACHE? Will this work if you rerun cmake with a different $ATDM_CONFIG_KOKKOS_ARCH?

@jjellio, they are defined here:

https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/utils/ATDMDevEnvUtils.cmake

Reply:
Thanks Ross, the comments are helpful.  I have a massacred ATDMDevSettings.cmake that allows for link time optimizations. (it requires twiddling CMAKE_AR and its default output).  I'd like to commit that (separately of course) - understanding the different ways to set variables will hopefully help me get that right the first time.